sklearn/manifold/t_sne.py

```diff
@@ -336,6 +348,11 @@ def _gradient_descent(objective, p0, it, n_iter,
 
     tic = time()
     for i in range(it, n_iter):
+        check_convergence = (i + 1) % n_iter_check == 0
+        # only compute the error when needed
+        if 'compute_error' in kwargs:
```

Comment:
why do we need this? What is the case where kwargs does not include `compute_error`? Do we need to be passing it in at all or just setting it here?

Reply:
It was to deal with [this test](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/manifold/tests/test_t_sne.py#L47-L53).
I though it was safer to put this safe guard, but I am not sure anymore since `_gradient_descent` is private

Reply:
How about just put it comute_error=True in the default kwargs set above when kwargs is None? Then the logic in the loop is clearer and independent of the call params.